src/main/java/org/opendatanode/plugins/transformer/exceltocsv/ExcelToCsv.java

```diff
@@ -0,0 +1,263 @@
+                    DpuFile csvFile = createCsvFile(entry.getSymbolicName(), sheet.getSheetName());
+                    ContextUtils.sendShortInfo(ctx, "ExcelToCsv.csvGenerationStarted", csvFile.symbolicName);
+                    sheetToCsv(sheet, csvFile.file);
+                    //addFileToOutput(csvFile); // method createCsvFile() added file to output (it was done by method FilesHelper.createFile())
+                    ContextUtils.sendShortInfo(ctx, "ExcelToCsv.csvGenerationFinished", csvFile.symbolicName);
+                } else {
+                    ContextUtils.sendShortInfo(ctx, "ExcelToCsv.sheetIgnored", sheet.getSheetName());
+                }
+            }
+        } catch (EncryptedDocumentException | InvalidFormatException | IOException ex) {
+            ContextUtils.sendError(ctx, "ExcelToCsv.excelTransformationFailed", ex.toString(), entry.getSymbolicName());
+            throw ex;
+        } finally {
+            if (wb != null) {
+                wb.close();
+            }
+        }
+
+        ContextUtils.sendShortInfo(ctx, "ExcelToCsv.excelTransformationFinishedSuccessfully", entry.getSymbolicName());
+    }
+
+    private DpuFile createCsvFile(String excelFileName, String sheetName) throws DataUnitException {
+        excelFileName = getBaseFileName(excelFileName); // removes file extension
+
+        String csvFileName = config.getCsvFileNamePattern().replace(ExcelToCsvConfig_V1.PLACEHOLDER_EXCEL_FILE_NAME,
+                excelFileName);
+        csvFileName = csvFileName.replace(ExcelToCsvConfig_V1.PLACEHOLDER_SHEET_NAME, sheetName);
+
+        Entry newEntry = FilesHelper.createFile(output, csvFileName);
+        File file = FilesHelper.asFile(newEntry);
+        return new DpuFile(csvFileName, file);
+    }
+
+    /**
+     * @param fileName File name, eg. example.xlsx
+     * @return Base file name (file name without extension).
+     */
+    private String getBaseFileName(String fileName) {
+        // for "example.xlsx" parts is ["example", "xlsx"]
+        String[] parts = fileName.split("\\.(?=[^\\.]+$)");
+        return parts[0];
+    }
+
+    private void addFileToOutput(DpuFile file) throws DataUnitException {
+        FilesHelper.addFile(output, file.file, file.symbolicName);
+    }
+
+    private void sheetToCsv(Sheet sheet, File csvFile) throws IOException {
+        final String[] empty_array = new String[]{};
+        CSVWriter csvWriter = null;
+        try {
+            csvWriter = new CSVWriter(new OutputStreamWriter(new FileOutputStream(csvFile), "utf-8"));
+
+            int minCellNumber = getMinCellNumber(sheet);
+            for (int r = sheet.getFirstRowNum(); r <= sheet.getLastRowNum(); r++) {
+                Row row = sheet.getRow(r);
+                if (row != null) {
+                    List<String> csvLine = new ArrayList<>();
+                    for (int c = minCellNumber; c < row.getLastCellNum(); c++) {
```

Comment:
Don't need to loop over all items in list. When we find first non-empty cell then we can stop.